client/src/client.rs

```diff
@@ -534,6 +498,53 @@ impl Client {
     }
 }
 
+/// Client implements a JSON-RPC client for the Bitcoin Core daemon or compatible APIs.
+pub struct Client {
+    client: jsonrpc::client::Client,
+}
+
+impl Client {
+    /// Creates a client to a bitcoind JSON-RPC server.
+    pub fn new(url: String, user: Option<String>, pass: Option<String>) -> Self {
+        debug_assert!(pass.is_none() || user.is_some());
+
+        Client {
+            client: jsonrpc::client::Client::new(url, user, pass),
+        }
+    }
+
+    /// Create a new Client.
+    pub fn from_jsonrpc(client: jsonrpc::client::Client) -> Client {
```

Comment:
This could be replaced with a `From` trait implementation.

Reply:
Correct, but I don't see this as a conversion. More like an instantiation.

Reply:
fair enough.